lucetc/src/output.rs

```diff
@@ -136,10 +136,15 @@ fn write_module(
     function_manifest_len: usize,
     obj: &mut Artifact,
 ) -> Result<(), Error> {
-    let mut native_data = Cursor::new(Vec::with_capacity(NATIVE_POINTER_SIZE * 4));
+    let mut native_data = Cursor::new(Vec::with_capacity(16 + NATIVE_POINTER_SIZE * 4));
```

Comment:
Does this have to be a magic number or can we use `size_of`?

Reply:
good point! the entire `with_capacity` argument can and should be `mem::size_of::<SerializedModule>` now that such a type exists.